src/basic/fd-util.c

```diff
@@ -1001,15 +1001,14 @@ int fd_vet_accmode(int fd, int mode) {
 int fd_is_writable(int fd) {
         int flags, mode;
 
-        if (fd < 0)
-                return -EBADF;
+        assert(fd >= 0);
 
         flags = fcntl(fd, F_GETFL);
         if (flags < 0)
                 return -errno;
 
         if (FLAGS_SET(flags, O_PATH))
-                return 0;
+                return false;
```

Comment:
Hmm, you mixed the commits... Please move the changes for fd-util.c to the first commit.